subsys/bluetooth/audio/csip_crypto.c

```diff
@@ -61,32 +61,30 @@ static void xor_128(const uint8_t a[16], const uint8_t b[16], uint8_t out[16])
 	}
 }
 
-int bt_csip_sih(const uint8_t sirk[BT_CSIP_SET_SIRK_SIZE], uint32_t r,
-		uint32_t *out)
+int bt_csip_sih(const uint8_t sirk[BT_CSIP_SET_SIRK_SIZE], uint8_t r[BT_CSIP_SIH_PRAND_SIZE],
+		uint8_t out[BT_CSIP_SIH_HASH_SIZE])
 {
-	uint8_t res[16]; /* need to store 128 bit */
+	uint8_t res[BT_CSIP_PRAND_SIZE]; /* need to store 128 bit */
 	int err;
 	uint8_t sirk_tmp[BT_CSIP_SET_SIRK_SIZE];
 
-	if ((r & BIT(23)) || ((r & BIT(22)) == 0)) {
-		LOG_DBG("Invalid r %0x06x", (uint32_t)(r & BT_CSIP_R_MASK));
+	if ((r[BT_CSIP_SIH_HASH_SIZE - 1] & BIT(7)) || ((r[BT_CSIP_SIH_HASH_SIZE - 1] & BIT(6)) == 0)) {
+		LOG_DBG("Invalid r %s", bt_hex(r, BT_CSIP_SIH_PRAND_SIZE));
 	}
 
 	LOG_DBG("SIRK %s", bt_hex(sirk, BT_CSIP_SET_SIRK_SIZE));
-	LOG_DBG("r 0x%06x", r);
+	LOG_DBG("r %s", bt_hex(r, BT_CSIP_SIH_PRAND_SIZE));
 
 	/* r' = padding || r */
 	(void)memset(res, 0, BT_CSIP_CRYPTO_PADDING_SIZE);
-	sys_put_be24(r, res + BT_CSIP_CRYPTO_PADDING_SIZE);
 
-	LOG_DBG("BE: r' %s", bt_hex(res, sizeof(res)));
+	sys_memcpy_swap(res + BT_CSIP_CRYPTO_PADDING_SIZE, r, BT_CSIP_SIH_PRAND_SIZE);
```

Comment:
This implies that `r` is stored as LE instead of system endianess, right? If that is the case, then the documentation for `bt_csip_sih` needs to be updated

Reply:
That is correct - however, the reason why I did not document it is because it is now stored in an array. An array is still an array, and the natural way of ordering it would be LSB at index 0, and MSB at index (len-1), wouldn't you agree?
Both a BE and LE system would use an array in the same way (unless you are using memcpy to store e.g. a uint32 in the array).

As far as I can tell, there is no mention of the ordering in the current sih documentation.

Reply:
> An array is still an array, and the natural way of ordering it would be LSB at index 0, and MSB at index (len-1), wouldn't you agree?
> Both a BE and LE system would use an array in the same way (unless you are using memcpy to store e.g. a uint32 in the array).

That is incorrect. If you have value 0x1234, then that will be store as 
```
array[0] = 0x34;
array[1] = 0x12;
```
in little endian, and 
```
array[0] = 0x12;
array[1] = 0x34;
```
in big endian. In little endian the least significant byte is stored at the lowest memory address, and in big endian it is the most significant byte. 

Take a look at the sys_put and sys_get functions which we have specifically to handle stuff like that.